src/Symfony/Component/Process/Process.php

```diff
@@ -1237,14 +1237,31 @@ public static function isPtySupported()
             return $result;
         }
 
-        if ('\\' === DIRECTORY_SEPARATOR) {
+        if (self::isWindowsPlatform()) {
             return $result = false;
         }
 
         return $result = (bool) @proc_open('echo 1', array(array('pty'), array('pty'), array('pty')), $pipes);
     }
 
     /**
+     * Returns whether the current process is run on Windows platform (Command Prompt, PowerShell) or not.
+     * 
+     * Keep in mind, that Windows users can use Unix shells (like the git-shell) with TTY/PTY support,
+     * in this case, this function will return 'false'.
+     *
+     * @return bool
+     */
+    public static function isWindowsPlatform()
```

Comment:
you are right, there are 3 different environments a process can run in:
* Windows
  * Command Prompt, PowerShell
  * Unix shell (git-bash)
* Unix

I assumed that the last two behave the same.
I would have to add a method like:
```
private static function isUnixOnWindowsPlatform()
{
        if (self::isWindowsPlatform() && file_exists('/dev/tty') && is_readable('/dev/tty')) {
            return true;
        }
        return false;
}
private static function isWindowsPlatform()
{
        if ('\\' === DIRECTORY_SEPARATOR) {
            return true;
        }
        return false;
}
```
correct me if I'm wrong.
but this would make the code a bit more complicated...